parser_library/src/workspaces/workspace.h

```diff
@@ -134,7 +134,7 @@ class workspace : public diagnosable_impl, public parse_lib_provider, public lsp
 private:
     constexpr static char FILENAME_PROC_GRPS[] = "proc_grps.json";
     constexpr static char FILENAME_PGM_CONF[] = "pgm_conf.json";
-    constexpr static char HLASM_PLUGIN_FOLDER[] = ".hlasmplugin";
+    constexpr static char HLASM_PLUGIN_FOLDER[] = ".hlasmplugin/";
```

Comment:
It seems strange to me that this change was required. What's the reason behind it?

Reply:
The reasoning behind this is the same which goes for the unintuitive `resource_location::join()`.

Trailing slashes in the URIs denote directories whereas URIs without the trailing slash represent files -> hence the suffix here. Other solution would be to use `resource_location::to_directory()` function instead which is fine as well.

As for the `resource_location::join()` function, when following the same logic, it didn't make sense to me to do the `join()` with a file. Therefore, the file part of the URI was cut.

Anyway, I'd like to hear your thoughts on both of these topics.

I hope to find a different source but it will need to wait until tomorrow:(https://stackoverflow.com/questions/5948659/when-should-i-use-a-trailing-slash-in-my-url#:~:text=Conventionally%2C%20a%20trailing%20slash%20(%2F),the%20URL%20was%20a%20file.)

Reply:
It turned out that this topic is not a straightforward as it would seem.

I've added some `lexically` functions and tried to mimic the behavior shown by `std::filesystem`. I've used this approach for `join()` as well so it behaves how you've suggested.

Also, `relative_reference_resolution()` based on RFC 3986 was added which behaves in a slightly similar way as was shown by the original `join` behavior (appending new path after last slash and not after last character).